test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/SearchTestsBase.cs

```diff
@@ -258,5 +282,150 @@ protected void ValidateBundleUrl(Uri expectedBaseAddress, ResourceType expectedR
 
             Assert.Equal(HttpUtility.UrlDecode(uriBuilder.Uri.ToString()), HttpUtility.UrlDecode(bundleUrl));
         }
+
+        // Troubleshooting helper methods follow.
+        // These methods can be used to debug test failures in the pipeline.
+        // ExecuteAndValidateBundle method is partially instrumented with WriteSearchAsync, which can be enabled by calling constructor with the output parameter set.
+        protected async System.Threading.Tasks.Task WriteSearchAsync(string title, string pathAndQuery, Tuple<string, string> header = null)
+        {
+            Exception exception = null;
+            FhirResponse<Bundle> fhirResponse = null;
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                fhirResponse = await Client.SearchAsync(pathAndQuery, header);
+            }
+            catch (Exception ex)
+            {
+                exception = ex;
+            }
+#pragma warning restore CA1031 // Do not catch general exception types
+
+            WriteSearchAsync(fhirResponse, title, pathAndQuery, header, exception);
+        }
+
+        protected void WriteSearchAsync<T>(FhirResponse<T> fhirResponse, string title, string requestPathAndQuery, Tuple<string, string> requestHeader = null, Exception exception = null)
+            where T : Resource
+        {
+            _output.WriteLine($"<--------- {title}");
+            _output.WriteLine($"REQUEST:");
+            _output.WriteLine($"  {requestPathAndQuery}");
+            if (requestHeader != null)
+            {
+                _output.WriteLine($"  {requestHeader.Item1}: {requestHeader.Item2}");
+            }
+
+            _output.WriteLine($"RESPONSE:");
+
+            if (fhirResponse != null)
+            {
+                _output.WriteLine($"  {fhirResponse.StatusCode}");
+                if (fhirResponse.Headers != null)
+                {
+                    string headerName = "X-Instance-Id";
```

Comment:
Perhaps add a new constant to the class KnownHeaders instead of hard coding the string.

Reply:
done